game/scripts/vscripts/modifiers/modifier_duel_invulnerability.lua

```diff
@@ -0,0 +1,20 @@
+modifier_duel_invulnerability = class({})
+
+function modifier_duel_invulnerability:CheckState()
+	return {
+	[MODIFIER_STATE_INVULNERABLE] = true,
+	[MODIFIER_STATE_MAGIC_IMMUNE] = true,
+	}
+end
+
+function modifier_duel_invulnerability:IsHidden() return true end
+function modifier_duel_invulnerability:IsPurgable()	return false end
+function modifier_duel_invulnerability:DestroyOnExpire() return true end
+
+
+function modifier_duel_invulnerability:OnCreated(kv) 
+	if IsServer() then
+		self.duration = kv.duration
+		self:SetDuration(self.duration, true)
```

Comment:
This is completely unnecessary. `duration` is a special key that will spawn the modifier with the set duration automatically.